src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Presentation/CompletionSet2.cs

```diff
@@ -101,13 +108,14 @@ internal void SetTrackingSpan(ITrackingSpan trackingSpan)
         private VSCompletion ConvertCompletionItem(CompletionItem item, string displayTextOpt = null)
         {
             VSCompletion value;
-            if (!_completionItemMap.TryGetValue(item, out value))
+            if (!_completionItemToVSCompletion.TryGetValue(item, out value))
             {
                 value = new CustomCommitCompletion(
                     _completionPresenterSession,
                     item,
                     displayTextOpt ?? item.DisplayText);
-                _completionItemMap.Add(item, value);
+                _completionItemToVSCompletion.Add(item, value);
+                value.Properties.AddProperty(typeof(CompletionItem), item);
```

Comment:
Is the property bag lazily created on `CompletionItem`?  From the comment below, I'm concerned about whether this will be an allocation hotspot.